libraries/logalyzer.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+/**
+ * Analyze the log of a Chronos job using predefined keywords
+ */
+class Logalyzer_Library {
+    private $job;
+    private $system;
+    private $log;
+    private $warningKeys;
+    private $errorKeys;
+
+    /**
+     * @throws Exception
+     */
+    public function __construct($job) {
+        $this->job = $job;
+        $this->system = new System($this->job->getSystemId());
+        $path = UPLOADED_DATA_PATH . '/log/' . $job->getId() . '.log';
+        $log = Util::readFileContents($path);
```

Comment:
This looks wrong. Do not read the whole file in the constructor, move to readEntireLog.